src/Polly.Core/Timeout/TimeoutResilienceStrategy.cs

```diff
@@ -83,11 +94,10 @@ protected internal override async ValueTask<Outcome<TResult>> ExecuteCore<TResul
 
     private static CancellationTokenRegistration CreateRegistration(CancellationTokenSource cancellationSource, CancellationToken previousToken)
     {
-        if (previousToken.CanBeCanceled)
-        {
-            return previousToken.Register(static state => ((CancellationTokenSource)state!).Cancel(), cancellationSource, useSynchronizationContext: false);
-        }
-
-        return default;
+#if NET
+        return previousToken.UnsafeRegister(static state => ((CancellationTokenSource)state!).Cancel(), cancellationSource);
```

Comment:
Same question here about `Unsafe` variant.

Reply:
This is also just avoiding any potential async state capture which isn't needed for the registration callback.